src/com/pty4j/unix/linux/OSFacadeImpl.java

```diff
@@ -93,8 +93,8 @@
     // Initialize architecture-specific constants
     String arch = System.getProperty("os.arch");
 
-    if (arch.equals("ppc64le")) {
-      // PowerPC64
+    if (arch.equals("ppc64le") || arch.equals("mips64el")) {
```

Comment:
This is strange. If I put the wrong TIOCGWINSZ and TIOCSWINSZ here, it will report an error in this line:
https://github.com/JetBrains/pty4j/blob/64dc275955205db420ceb8b1f2f6f468f3db0c07/src/com/pty4j/unix/Pty.java#L127
But, the terminal interface in IDE can be opened normally, only the number of rows and columns is not normal.

However, if I write the correct TIOCGWINSZ and TIOCSWINSZ, as it is now, the terminal interface in the IDE cannot be opened, and there are no errors at the same time.

I tnink I still need some help to make this part work properly under mips.

Sorry for my poor English.

Reply:
@wangling12 The mechanism of getting/setting window size is changed by 55de04da92b79b8356959268a785c72f24f354d6. Now `TIOCGWINSZ`/`TIOCSWINSZ` constants are not needed on Java side. Please update, rebuild libpty.so and try again. Will it help? If no, please provide more details about problem.

Do you know if it's possible to build libpty.so for mips using Linux x64_86? I'd like to be able to build it on my side.

Reply:
No improvement, same as before.
When I click on the terminal, the terminal window will pop up, but it will close quickly.

![截图录屏_选择区域_20201102120125](https://user-images.githubusercontent.com/24770600/97829100-dd4f4800-1d03-11eb-8bbe-98f16036c664.png)
![截图录屏_选择区域_20201102120139](https://user-images.githubusercontent.com/24770600/97829104-e3452900-1d03-11eb-8ded-b82e341d26c4.png)


Terminal log at the same time:
```shell
./idea.sh
OpenJDK 64-Bit Server VM warning: You have loaded library /usr/lib/mips64el-linux-gnuabi64/jni/libatk-wrapper.so.6.0.0 which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
2020-11-02 12:01:03,293 [   4984]   WARN - openapi.util.io.FileSystemUtil - Failed to load filesystem access layer: Linux, 1.8.0 
java.lang.IllegalStateException: Unsupported OS/arch: Linux/mips64el
	at com.intellij.openapi.util.io.FileSystemUtil$JnaUnixMediatorImpl.<init>(FileSystemUtil.java:293)
	at com.intellij.openapi.util.io.FileSystemUtil.getMediator(FileSystemUtil.java:58)
	at com.intellij.openapi.util.io.FileSystemUtil.<clinit>(FileSystemUtil.java:49)
	at com.intellij.openapi.vfs.newvfs.persistent.FSRecords.<clinit>(FSRecords.java:80)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.<init>(PersistentFSImpl.java:94)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.intellij.serviceContainer.ConstructorInjectionKt.instantiateUsingPicoContainer(constructorInjection.kt:47)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClassWithConstructorInjection(ComponentManagerImpl.kt:664)
	at com.intellij.serviceContainer.ServiceComponentAdapter.createAndInitialize(ServiceComponentAdapter.kt:49)
	at com.intellij.serviceContainer.ServiceComponentAdapter.doCreateInstance(ServiceComponentAdapter.kt:37)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstanceUncached(BaseComponentAdapter.kt:113)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstance(BaseComponentAdapter.kt:67)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstance$default(BaseComponentAdapter.kt:60)
	at com.intellij.serviceContainer.ComponentManagerImpl$preloadServices$future$1.run(ComponentManagerImpl.kt:821)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1626)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:215)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:26)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:194)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:186)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:652)
	at java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:649)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:649)
	at java.lang.Thread.run(Thread.java:748)
2020-11-02 12:01:05,508 [   7199]   WARN - tellij.ide.SystemHealthMonitor - issue detected: outdated.jre.version.message2 (ignored) 

** (java:15966): WARNING **: 12:01:07.789: jaw_impl_get_instance called from jaw_thread. If you are running a screen reader, this is expected
If you are not running a screen reader, please report this warning to the java-atk-wrapper package, explaining how to reproduce this warning


2020-11-02 12:01:09,623 [  11314]   WARN - Container.ComponentManagerImpl - Do not use constructor injection (requestorClass=org.jetbrains.android.compose.AndroidComposeAutoDocumentation) 
2020-11-02 12:01:12,333 [  14024]   WARN - tartup.impl.StartupManagerImpl - Activities registered via registerPostStartupActivity must be dumb-aware: org.jetbrains.kotlin.idea.configuration.ui.KotlinConfigurationCheckerComponent$projectOpened$1@4dcc21e7 
2020-11-02 12:01:18,958 [  20649]   WARN - com.intellij.util.xmlb.Binding - no accessors for org.jetbrains.kotlin.idea.highlighter.KotlinDefaultHighlightingSettingsProvider 

** (java:15966): WARNING **: 12:01:46.956: jaw_impl_get_instance called from jaw_thread. If you are running a screen reader, this is expected
If you are not running a screen reader, please report this warning to the java-atk-wrapper package, explaining how to reproduce this warning


2020-11-02 12:01:48,635 [  50326]   WARN - com.intellij.util.xmlb.Binding - no accessors for org.jetbrains.kotlin.idea.core.script.configuration.utils.ScriptClassRootsStorage 
```

And testInitialColumnsAndRows test failed. 

```shell
junit.framework.ComparisonFailure: expected:<[columns: 111, rows: 11]
> but was:<[OpenJDK 64-Bit Server VM warning: You have loaded library /home/uos/.cache/JNA/temp/jna3701149388392047802.tmp which might have disabled stack guard. The VM will try to fix the stack guard now.]
>
	at junit.framework.Assert.assertEquals(Assert.java:100)
	at junit.framework.Assert.assertEquals(Assert.java:107)
	at junit.framework.TestCase.assertEquals(TestCase.java:269)
	at com.pty4j.PtyTest.testInitialColumnsAndRows(PtyTest.java:216)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:106)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:66)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:117)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:155)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:137)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
I have no idea about build libpty.so for mips using Linux x64_86. I'm working on a MIPS machine, but I'm not sure whether the x86 to MIPS cross compiler toolchain is reliable.

Reply:
>When I click on the terminal, the terminal window will pop up, but it will close quickly.

@wangling12 Please attach your idea.log (Help | Show Log in ...).

>And testInitialColumnsAndRows test failed.

It should be fixed now, please try again.

I've set up Debian MIPS in AWS, but no luck so far - `./gradlew tasks` fails when doing SSL handshake (using openjdk-11-jdk). What Java version do you have installed on your MIPS machine? And how you installed it? (debian package, something else?)